### PR TITLE
Remove Visual Studio warnings about unresolvable references

### PR DESCRIPTION
**Pull request by JanLenoch**
Probably due to some recent breaking changes in the framework, VS started showing the MSB3245 compiler warnings. It warned that some of the framework assemblies were not found in their expected filesystem locations.

Although the exact cause is not known, trying to raise the `TargetFramework` to `netstandard2.0` resolved the issue. The upgrade did not incur any code changes, only the .csproj and AssemblyInfo.cs were affected. All other code changes deal with comments in #115 .

I've also upgraded the dependencies to their newest versions.

### CODE REVIEW

**Comment by petrsvihlik** [comment]
@JanLenoch fix the failing builds pls

**Comment by JanLenoch** [comment]
The comments from the review were incorporated via a new [branch](https://github.com/Kentico/delivery-sdk-net/tree/100-unresolvable-references). The code runs correctly in local environments but not in AppVeyor. I'd need to consult it with someone versed with AV.

File: KenticoCloud.Delivery/DeliveryClient.cs [hunk 1]
```diff
@@ -522,23 +522,22 @@ public DeliveryClient(string projectId, string previewApiKey) : this(new Deliver
...
-                }
-            );
-
-            return await GetResponseContent(policyResult?.FinalHandledResult ?? policyResult?.Result);
-        }
-
-        private async Task<JObject> GetSolitaryDeliveryResponseAsync(string endpointUrl)
-        {
-            return await GetResponseContent(await SendHttpMessage(endpointUrl));
+            if (_deliveryOptions.EnableResilienceLogic)
```
Comment (petrsvihlik) [inline KenticoCloud.Delivery/DeliveryClient.cs]: why do I see retry logic-related code in this PR? is that intentional?

File: KenticoCloud.Delivery.Rx/KenticoCloud.Delivery.Rx.csproj [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
     <RepositoryType>git</RepositoryType>
-    <PackageTags>kentico;mvc;aspnet;aspnetmvc;rx;reactive;observable;linq;events</PackageTags>
+    <PackageTags>kentico;mvc;aspnet;aspnetmvc;reactive;rx</PackageTags>
+    <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Newtonsoft.Json" Version="9.0.1" />
-    <PackageReference Include="System.Reactive" Version="3.1.1" />
+    <PackageReference Include="Newtonsoft.Json" Version="11.0.2" />
```
Comment (petrsvihlik) [inline KenticoCloud.Delivery.Rx/KenticoCloud.Delivery.Rx.csproj]: why are these updates necessary? do we really need to update them in order to be compliant with netstandard2.0? do we want to force our consumers to use these latest versions?

File: KenticoCloud.Delivery/DeliveryClient.cs (hunk 1 above)
Comment (JanLenoch) [reply KenticoCloud.Delivery/DeliveryClient.cs]: Yes, it is. The PR goes from the 100-unresolvable-references branch to the 100-retry-policy branch. The changes are still closely related to the retry policy.

File: KenticoCloud.Delivery.Rx/KenticoCloud.Delivery.Rx.csproj (hunk 2 above)
Comment (JanLenoch) [reply KenticoCloud.Delivery.Rx/KenticoCloud.Delivery.Rx.csproj]: I know that the 9.0.1 version of the Newtonsoft.Json package was part of the reason why my Blazor app could not use the Delivery SDK. I'll find the lowest working version.

Comment (JanLenoch) [reply KenticoCloud.Delivery.Rx/KenticoCloud.Delivery.Rx.csproj]: I've chosen the most-downloaded versions. Json v10.0.3: 18600k downloads. Reactive v3.1.1: 1124k downloads.

**Comment by JanLenoch** [comment]
SDK tested in both an ASP.NET Core 2.1 app and an MVC 5 app (.NET Framework 4.5).

**Pull request merged**